test/cpp/tensorexpr/test_loopnest.cpp

```diff
@@ -1706,5 +1706,260 @@ void testUnrollWithLet() {
+  KernelScope kernel_scope;
+
+  // Input IR:
+  //   for (int x = -50; x < 100; x++) {
+  //     A[x + 50] = B[x + 50];
+  //     B[x + 50] = x * 2;
+  //   }
+  const int kTotalSize = 150;
+  BufHandle a_buf("A", {ExprHandle(kTotalSize)}, kInt);
+  BufHandle b_buf("B", {ExprHandle(kTotalSize)}, kInt);
+  VarHandle x("x", kInt);
+  auto for_body = Block::make(
+      {Store::make(a_buf, {x + 50}, Load::make(kInt, b_buf, {x + 50}, 1), 1),
+       Store::make(b_buf, {x + 50}, x * 2)});
+  auto for_stmt = For::make(x, -50, 100, for_body);
+  Block::make({for_stmt});
+
+  For* normalized = nullptr;
+  LoopNest::normalize(for_stmt, &normalized);
+
+  std::ostringstream oss;
+  oss << *normalized;
+  const std::string& expected_ir =
+      R"IR(
+        # CHECK: for (int x = 0; x < 100 - -50; x++) {
+        # CHECK:   A[(x + -50) + 50] = B[(x + -50) + 50];
+        # CHECK:   B[(x + -50) + 50] = (x + -50) * 2;
+      )IR";
+  torch::jit::testing::FileCheck().run(expected_ir, oss.str());
+}
+
+void testNormalizeStartZero() {
+  KernelScope kernel_scope;
+
+  // Input IR:
+  //   for (int x = 0; x < 100; x++) {
+  //     A[x] = B[x];
+  //     B[x] = x * 2;
+  //   }
+  // Should not be modified.
+
+  const int kTotalSize = 100;
+  BufHandle a_buf("A", {ExprHandle(kTotalSize)}, kInt);
+  BufHandle b_buf("B", {ExprHandle(kTotalSize)}, kInt);
+  VarHandle x("x", kInt);
+  auto for_body =
+      Block::make({Store::make(a_buf, {x}, Load::make(kInt, b_buf, {x}, 1), 1),
+                   Store::make(b_buf, {x}, x * 2)});
+  auto for_stmt = For::make(x, 0, 100, for_body);
+  Block::make({for_stmt});
+
+  For* normalized = nullptr;
+  LoopNest::normalize(for_stmt, &normalized);
+
+  std::ostringstream oss;
+  oss << *normalized;
+  const std::string& expected_ir =
+      R"IR(
+        # CHECK: for (int x = 0; x < 100; x++) {
+        # CHECK:   A[x] = B[x];
+        # CHECK:   B[x] = x * 2;
+      )IR";
+  torch::jit::testing::FileCheck().run(expected_ir, oss.str());
+}
+
+void testNormalizeStartVariable() {
+  KernelScope kernel_scope;
+
+  // Input IR:
+  //   for (int x = y; x < 100; x++) {
+  //     A[x] = B[x];
+  //     B[x] = x * 2;
+  //   }
+  // Should not be modified.
+
+  const int kTotalSize = 100;
+  BufHandle a_buf("A", {ExprHandle(kTotalSize)}, kInt);
+  BufHandle b_buf("B", {ExprHandle(kTotalSize)}, kInt);
+  VarHandle x("x", kInt);
+  VarHandle y("y", kInt);
+  auto for_body =
+      Block::make({Store::make(a_buf, {x}, Load::make(kInt, b_buf, {x}, 1), 1),
+                   Store::make(b_buf, {x}, x * 2)});
+  auto for_stmt = For::make(x, y, 100, for_body);
+  Block::make({for_stmt});
+
+  For* normalized = nullptr;
+  LoopNest::normalize(for_stmt, &normalized);
+
+  std::ostringstream oss;
+  oss << *normalized;
+  const std::string& expected_ir =
+      R"IR(
+        # CHECK: for (int x = 0; x < 100 - y; x++) {
+        # CHECK:   A[x + y] = B[x + y];
+        # CHECK:   B[x + y] = (x + y) * 2;
+      )IR";
```

Comment:
Let's think about this - normalize in this instance is making the resultant graph more complex and thats probably not what the user wants. I think it would be reasonable to bail if the start index was not constant.